arch/Kconfig

```diff
@@ -314,6 +343,12 @@ config BOOTLOADER_CONTEXT_RESTORE_SUPPORTED
 # End hidden CPU family configs
 #
 
+config CPU_HAS_SAU
+	bool
+	help
+	  This option is enabled when the CPU has
+	  a Security Attribution Unit (SAU).
+
```

Comment:
maybe this is too ARM specific?

Reply:
> 
> 
> maybe this is too ARM specific?

I thought, also, about this. My intention was not to make it look like the ARM SAU. However, "Security Attribution" is a rather generic name, IMHO. 

@ruuddw: would CPU_HAS_SAU fit the nomenclature of ARC and SecureShield in your opinion?

Actually, I have no strong opinion on how to name this generic Kconfig option, as long as it describes that the CPU has capabilities for assigning security attributions.

Reply:
Why not use CPU_HAS_TEE? At the end of the day havin a SAU essentially amounts to supporting trusted execution, does it not?

Reply:
That is also a possibility (CPU_HAS_TEE, or CPU_HAS_TRUSTED_EXECUTION)

Reply:
@carlescufi I changed this symbol to CPU_HAS_TEE, and updated the help text accordingly.

Reply:
ARC SecureShield doesn't use SAU or other "secure attribution unit", I prefer the generic TEE or Trusted Execution variant.
What's the difference between ARCH_HAS_TRUSTED_EXECUTION and CPU_HAS_TEE? Aren't these the same?

Reply:
The CPU_HAS_MPU and ARCH_HAS_STACK_PROTECTION is different, indicating different concepts. I understand the difference you try to indicate, but is it relevant/needed or just adding config complexity? Is there a use-case where they have different values? And there's the ARM_TRUSTZONE_AS well, still feels like 3 config options indicating the same. At most 2 seems enough: one at user level to enable disable TEE support, and (if other options can't be used for that) indicating whether the hardware has the required features or not to enable it.

Reply:
>Is there a use-case where they have different values?

Yes, there is. I explain here:

In Cortex-M23 CPUs, the existence of the Stack Pointer Limit registers (MSPLIM, PSPLIM) depends on the existence of Security Extension, but, itself, the Stack Pointer Limit checking is _not_ a Trusted Execution feature!!

So, CPU_HAS_TEE is a "pure" HW-cabability. We must set it in Cortex-M23 with Security Extension, in order to use the built-in stack overflow protection scheme in Cortex- M23.

However, we might _not_ care at all about Trusted Execution. Therefore, we have the user option at Board level (CONFIG_ARM_TRUSTZONE_M), to select or not TrustZone support.

CONFIG_ARM_TRUSTZONE_M will "select" ARCH_HAS_TRUSTED_EXECUTION, and the latter will enable the setting of TRUSTED_EXECUTION_(NON)SECURE.

Reply:
My main point is that we agree on the generic, architecture neutral names. ARCH_HAS_TRUSTED_EXECUTION (indicating the capability of the platform) and TRUSTED_EXECUTION_(NON)SECURE (as user selectable option to indicate enabling/using the capability) is fine. CPU_HAS_TEE sounds generic as well, but currently has only local 'ARM' meaning (e.g. all ARC processors can have stack pointer checking, it is not restricted to SecureShield/CPU_HAS_TEE).